packages/app-desktop/ElectronAppWrapper.ts

```diff
@@ -260,7 +322,8 @@ export default class ElectronAppWrapper {
 					this.hide();
 				}
 			} else {
-				if (this.trayShown() && !this.willQuitApp_) {
+				const hasBackgroundWindows = this.secondaryWindows_.size > 0;
+				if ((hasBackgroundWindows || this.trayShown()) && !this.willQuitApp_) {
 					event.preventDefault();
 					this.win_.hide();
```

Comment:
This allows the main window to be closed without quitting the app, even if "show tray icon" is disabled.